app/controllers/users_controllers.rb

```diff
@@ -0,0 +1,35 @@
+class UsersControllers < ApplicationController
+  before_action :require_admin, only: [:index, :show, :new, :create, :destroy]
+  before_action :require_user, only: [:edit, :update]
+
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = User.find(params[:id])
```

Comment:
You are using @user = User.find(params[:id]) this 3 times in file.
There are two different options how to cope with that: before action and memoization.
before_action: set_user
def set_user
  @user=...
end

def user
@user ||= ...
end
Feel free to ping if you need more info